homeassistant/components/devolo_home_control/binary_sensor.py

```diff
@@ -0,0 +1,80 @@
+"""Platform for binary sensor integration."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN
+from .devolo_device import DevoloDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Get all binary sensor and multi level sensor devices and setup them via config entry."""
+    entities = []
+
+    for device in hass.data[DOMAIN]["homecontrol"].binary_sensor_devices:
+        for binary_sensor in device.binary_sensor_property:
+            entities.append(
+                DevoloBinaryDeviceEntity(
+                    homecontrol=hass.data[DOMAIN]["homecontrol"],
+                    device_instance=device,
+                    element_uid=binary_sensor,
+                )
+            )
+
+    async_add_entities(entities, False)
+
+
+class DevoloBinaryDeviceEntity(DevoloDeviceEntity, BinarySensorEntity):
+    """Representation of a binary sensor within devolo Home Control."""
+
+    def __init__(self, homecontrol, device_instance, element_uid):
+        """Initialize a devolo binary sensor."""
+        if device_instance.binary_sensor_property.get(element_uid).sub_type != "":
+            name = (
+                device_instance.itemName
+                + " "
+                + device_instance.binary_sensor_property.get(element_uid).sub_type
+            )
+        else:
+            name = (
+                device_instance.itemName
+                + " "
+                + device_instance.binary_sensor_property.get(element_uid).sensor_type,
+            )
```

Comment:
```suggestion
        element = device_instance.binary_sensor_property.get(element_uid)
        if device_instance.binary_sensor_property.get(element_uid).sub_type != "":
            name = f"{device_instance.itemName} {element.sub_type}"
        else:
            name = f"{device_instance.itemName} {element.sensor_type}"
```

Almost there.  The standard is to use f-strings instead of concatenating strings when possible.

Reply:
Finally ;)
Done with latest commit